webapp/tests/karma/ts/modules/reports/reports.component.spec.ts

```diff
@@ -183,55 +199,160 @@ describe('Reports Component', () => {
     });
 
-    it('should add selected report when in select mode and not already selected with some selected reports', () => {
+    it('should add selected report if not already selected when there are some selected reports', () => {
       component.selectMode = true;
       component.selectedReports = [{ _id: 'selected1' }, { _id: 'selected2' }];
 
-      component.toggleSelected({ _id: 'rid' });
+      component.selectReport({ _id: 'rid' });
+
       expect(addSelectedReport.callCount).to.equal(1);
       expect(addSelectedReport.args[0]).to.deep.equal([{ _id: 'rid' }]);
       expect(selectReport.callCount).to.equal(1);
       expect(selectReport.args[0]).to.deep.equal([{ _id: 'rid' }]);
       expect(removeSelectedReport.callCount).to.equal(0);
     });
 
-    it('should remove selected report if in select mode and already selected', () => {
+    it('should remove selected report if already selected', () => {
       component.selectMode = true;
       component.selectedReports = [{ _id: 'selected1' }, { _id: 'selected2' }, { _id: 'rid' }];
 
-      component.toggleSelected({ _id: 'rid' });
+      component.selectReport({ _id: 'rid' });
       expect(addSelectedReport.callCount).to.equal(0);
       expect(selectReport.callCount).to.equal(0);
       expect(removeSelectedReport.callCount).to.equal(1);
-      expect(removeSelectedReport.args[0]).to.deep.equal([{ _id: 'rid' }]);
+      expect(removeSelectedReport.args[0]).to.deep.equal([ { _id: 'rid' }, undefined ]);
+    });
+  });
+
+  describe('toggleAllSelected', () => {
+    let setSelectModeStub;
+    let deselectAllStub;
+    let selectAllStub;
+
+    beforeEach(() => {
+      setSelectModeStub = sinon.stub(ReportsActions.prototype, 'setSelectMode');
+      deselectAllStub = sinon.stub(ReportsActions.prototype, 'deselectAll');
+      selectAllStub = sinon.stub(ReportsActions.prototype, 'selectAll');
+    });
+
+    it('should select all when not all reports have been selected yet', () => {
+      component.selectedReports = [{ _id: 'selected1' }, { _id: 'selected2' }];
+      component.reportsList = [{ _id: 'selected1' }, { _id: 'selected2' }, { _id: 'selected3' }];
+
+      component.toggleAllSelected();
+
+      expect(selectAllStub.calledOnce).to.be.true;
+      expect(setSelectModeStub.notCalled).to.be.true;
+      expect(deselectAllStub.notCalled).to.be.true;
+    });
+
+    it('should deselect all when all reports have been selected', () => {
+      component.selectedReports = [{ _id: 'selected1' }, { _id: 'selected2' }, { _id: 'selected3' }];
+      component.reportsList = [{ _id: 'selected1' }, { _id: 'selected2' }, { _id: 'selected3' }];
+
+      component.toggleAllSelected();
+
+      expect(selectAllStub.notCalled).to.be.true;
+      expect(setSelectModeStub.calledOnce).to.be.true;
+      expect(deselectAllStub.calledOnce).to.be.true;
+    });
+  });
+
+  describe('bulkDeleteReports', () => {
+    it('should do not open modal when there are not selected reports', () => {
```

Comment:
typo: "should do not open modal when there are not selected reports"
```suggestion
    it('should not open modal when there are no selected reports', () => {
```

Reply:
ty!